plans/util/retrieve_and_upload.pp

```diff
@@ -29,22 +29,41 @@
     |-HEREDOC
     # lint:endignore
 
+$operating_system = run_task('peadm::os_identification', 'local://localhost')
+$os_string = $operating_system.first.value['osfamily']
+
+if os_string  == 'windows' {
+  $exists = run_command("[System.IO.File]::Exists('${local_path}')", 'local://localhost')
+  if $exists.first['stdout'].chomp == 'false' {
+    run_task('peadm::download', 'local://localhost',
+      source => $source,
+      path   => $local_path,
+    )
+  }
+} else {
   $exists = without_default_logging() || {
     run_command("test -e '${local_path}'", 'local://localhost',
       _catch_errors => true,
     ).ok()
   }
-
   unless $exists {
     run_task('peadm::download', 'local://localhost',
       source => $source,
       path   => $local_path,
     )
   }
+}
 
+if os_string  == 'windows' {
+  $result_size = run_task('peadm::filesize', 'local://localhost',
+    path => $local_path,
+  )
+  $local_size = $result_size['size']
+} else {
   $local_size = run_task('peadm::filesize', 'local://localhost',
     path => $local_path,
   ).first['size']
+}
```

Comment:
Can we merge this into one if/else block? If windows do all of this or else do all of this. I can see less repeated code if we do so, just the `download` block will repeat.

Reply:
Yes, thanks, should have caught this